sdk/analytics/src/main/java/com/klaviyo/analytics/Klaviyo.kt

```diff
@@ -60,8 +60,15 @@ object Klaviyo {
      * @return Returns [Klaviyo] for call chaining
      */
     fun setProfile(profile: Profile): Klaviyo = apply {
-        resetProfile()
-        UserInfo.updateFromProfile(profile)
+        if (UserInfo.isIdentified) {
```

Comment:
Heres the key piece: if currently just anonymous, we should still be merging
Also below I just figured since I'm setting userinfo var directly in the individual `setEmail` etc methods, I'd do the same here.

Reply:
Yeah I was gonna ask to what extent should we explain anonymous profiles to the app developers?

Reply:
Updated the comment without going into the weeds on what anonymous profiles are -- we can decide if we want a section about that in docs later